src/response.rs

```diff
@@ -136,9 +139,93 @@ impl Response {
             ext: Extensions::new(),
             peer_addr: None,
             local_addr: None,
+            // XXX(Jeremiah): should this be autogenerated on 4xx and 5xx codes?
+            error: None,
         }
     }
 
+    /// Create a new response from an `http_types::Error`.
+    ///
+    /// This will store the error in the `Response`, allowing it to later be
+    /// checked via `Response::error()`.
+    ///
+    /// If the `Error`'s status had a status code that was not a 5XX server
+    /// error code, the response body will be set to the error's message, and
+    /// the content-type header will be set to `http_types::mime::Plain`.
+    #[cfg(not(feature = "unstable"))]
+    pub fn from_error(error: Error) -> Self {
+        // Only send the message if it is a non-500 range error. All
+        // errors default to 500 by default, so sending the error
+        // body is opt-in at the call site.
+        let was_server_error = error.status().is_server_error();
+        let body = if !was_server_error {
+            Body::from_string(error.to_string())
```

Comment:
Why are we generating bodies at all here? Later middleware can generate a body if that's what the application needs, and a plaintext body isn't necessarily a particularly useful default. If we want a default response body for errors, I think it makes more sense to add a middleware that generates an error body based on the request's Accept header or something like that. I think this should just set the error and status

Reply:
This is taken from tide -- basically with hopes of consolidating this: https://github.com/http-rs/tide/pull/546/files#diff-3dbd16223f25e8b5e0bcf099743107e9R10-R23

Perhaps that's not quite the right move but it seemed fitting enough.

Reply:
I think this PR obviates that approach. If a middleware has not set a response body, the app doesn't need a response body

Reply:
After this change, #564 will just be a matter of the cors middleware setting a header on the response regardless of whether it has an Error, since https://github.com/http-rs/tide/pull/546/files#diff-dc3595a3d4ca6ef52648b5c158a0e100R164 will already be a Response

Reply:
Hmm, I'm not sure. Does this hurt? This allows to default case to be handled automatically if the user has not set a body or headers, and doesn't need some fancy dirty-checking in tide or surf to check if those should set these after middleware.

Reply:
This would be set *before* the error handling middleware though, which would result in allocating an unused body. The error handling middleware would have to check if it's a default body or an app-set body. I don't see any advantage to having default body behavior, since almost every production app will need to replace it. If there's default behavior, I think it should be implemented in tide, as a default middleware. This seems like an unnecessary exception to an otherwise simple-to-explain pattern.